app.py

```diff
@@ -182,6 +182,25 @@ def login_():
     def logout_():
         return logout()
 
+    @app.route('/push-log', methods=['POST'])
+    def push_log():
+        if 'level' not in request.json or 'message' not in request.json:
+            raise ValueError('Request body missing one or more mandatory fields ["message", "level"]')
+
+        _json = request.json.get
+        level, message, extra = _json('level'), _json('message'), _json('extra')
+        if not (extra is None or type(extra) is dict):
+            raise ValueError('Extra param must be a valid json object')
```

Comment:
question: This looks good to me for the time being, however: how you looked at done-for-you options to validate endpoint inputs?

Reply:
I've done some research, Flask doesn't provide you with anything unfortunately, but I'm working on an idea on how to add input validation in a uniform manner
I hope to be able to bring this in the near future